x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/DataTypes.java

```diff
@@ -60,160 +117,49 @@ public static DataType fromJava(Object value) {
-            // ODBC SQL_CODE_TIMESTAMP
-            return Integer.valueOf(3);
-        }
-        // ODBC null
-        return 0;
+    public static boolean isUnsupported(DataType from) {
+        return from == UNSUPPORTED;
     }
 
-    public static Short metaSqlMinimumScale(DataType t) {
-        return metaSqlSameScale(t);
+    public static boolean isString(DataType t) {
+        return t == KEYWORD || t == TEXT;
     }
 
-    public static Short metaSqlMaximumScale(DataType t) {
-        return metaSqlSameScale(t);
+    public static boolean isPrimitive(DataType t) {
+        return t != OBJECT && t != NESTED && t != UNSUPPORTED;
     }
 
-    // https://docs.microsoft.com/en-us/sql/odbc/reference/appendixes/decimal-digits
-    // https://github.com/elastic/elasticsearch/issues/40357
-    // since the scale is fixed, minimum and maximum should return the same value
-    // hence why this method exists
-    private static Short metaSqlSameScale(DataType t) {
-        // TODO: return info for SCALED_FLOATS (should be based on field not type)
-        if (t.isInteger()) {
-            return Short.valueOf((short) 0);
-        }
-        if (t.isDateBased() || t.isRational()) {
-            return Short.valueOf((short) t.defaultPrecision);
-        }
-        return null;
+    public static boolean isNull(DataType t) {
+        return t == NULL;
     }
 
-    // https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlgettypeinfo-function
-    public static Integer metaSqlRadix(DataType t) {
-        // RADIX  - Determines how numbers returned by COLUMN_SIZE and DECIMAL_DIGITS should be interpreted.
-        // 10 means they represent the number of decimal digits allowed for the column.
-        // 2 means they represent the number of bits allowed for the column.
-        // null means radix is not applicable for the given type.
-        return t.isInteger() ? Integer.valueOf(10) : (t.isRational() ? Integer.valueOf(2) : null);
+    public static boolean isNullOrNumeric(DataType t) {
+        return t.isNumeric() || isNull(t);
     }
 
-    //https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlgettypeinfo-function#comments
-    //https://docs.microsoft.com/en-us/sql/odbc/reference/appendixes/column-size
-    public static Integer precision(DataType t) {
-        if (t.isNumeric()) {
-            return t.defaultPrecision;
-        }
-        return t.displaySize;
+    public static boolean isSigned(DataType t) {
+        return t.isNumeric();
     }
 
-    public static boolean areTypesCompatible(DataType left, DataType right) {
+    public static boolean areCompatible(DataType left, DataType right) {
         if (left == right) {
             return true;
         } else {
             return
-                (left == DataType.NULL || right == DataType.NULL)
-                    || (left.isString() && right.isString())
+                (left == NULL || right == NULL)
+                    || (isString(left) && isString(right))
```

Comment:
`DataType` contains only methods that can work on its properties (such as whether it's numeric or not) while everything else is on `DataTypes`.
From the usage point of view it would be cleaner to have all methods inside one place, namely `DataTypes` (as new types can have new properties such as `interval`) but in case of numeric properties these are parts of the `DataType` itself.
This can be revisited in the future so that these properties are fully externalized (and essentially hard-coded just like `isString`).